src/components/event_view/event_info/PlanInfo.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+import EventField from '../event_fields/EventField';
+import EventFieldEdit from '../event_fields/EventFieldEdit';
+import {
+  usdFormatter,
+  convertToNumber
+} from '../../../shared/currencyFormatter';
+import { formatDateForMySql } from '../../../shared/dateFormatter';
+
+export default class PlanInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = this.getState(this.props.plan);
+    this.getPlan = this.getPlan.bind(this);
+    this.onClick = this.onClick.bind(this);
+    this.getField = this.getField.bind(this);
+    this.reserveEditMode = this.reserveEditMode.bind(this);
+  }
+
+  /**************************** Helper functions ****************************/
+
+  /**
+   * Receives and formats the values of a plan event from props.
+   * @param plan
+   * @returns {{method: string, loc_begin: string, loc_end: *,
+   * begin_time: string, end_time: string, cost: string, dscript: string}}
+   */
+  getState(plan) {
+    return {
+      loc: {
+        name: 'loc',
+        type: 'Location',
+        value: plan.loc ? plan.loc : 'unspecified',
+        editMode: false
+      },
+      begin_time: {
+        name: 'begin_time',
+        type: 'Begins',
+        value: plan.begin_time.toString(),
+        editMode: false
+      },
+      end_time: {
+        name: 'end_time',
+        type: 'Ends',
+        value: plan.end_time.toString(),
+        editMode: false
+      },
+      cost: {
+        name: 'cost',
+        type: 'Cost',
+        value: plan.cost ? usdFormatter.format(plan.cost) : 'unspecified',
+        editMode: false
+      },
+      dscript: {
+        name: 'dscript',
+        type: 'Description',
+        value: plan.dscript ? plan.dscript : 'unspecified',
+        editMode: false
+      }
+    };
+  }
+
+  /**
+   * Gets the plan.
+   */
+  getPlan() {
+    return {
+      e_id: this.props.plan.e_id,
+      trip_id: this.props.plan.trip_id,
+      loc: this.state.loc.value,
+      begin_time: formatDateForMySql(this.state.begin_time.value),
+      end_time: formatDateForMySql(this.state.end_time.value),
+      cost: convertToNumber(this.state.cost.value),
+      dscript: this.state.dscript.value,
+      completed: 0,
+      priority: 0
+    };
+  }
```

Comment:
Unless I'm mistaken, `PlanInfo` is the same as `AccomInfo`. Could you just have one class and pass two methods as properties: that is, `getEvent()` (which would have the implementation for `getPlan()` or `getAccom()`) and `getState(event)` (which would have specific implementation details for that type of event)?

Reply:
You're correct. We could do this, I like that idea. So, are you thinking we would then have two info files ```EventInfo``` and ```TranInfo```? The only differentiating feature between ```TranInfo``` and a general ```EventInfo``` file would be how many ```getField()``` calls are made i.e. 

```
render() {
    return (
      <div>
        <Table className="table-hover">
          <tbody>
            <tr>{this.getField(this.state.method)}</tr>
            <tr>{this.getField(this.state.loc_begin)}</tr>
            <tr>{this.getField(this.state.loc_end)}</tr>
            <tr>{this.getField(this.state.begin_time)}</tr>
            <tr>{this.getField(this.state.end_time)}</tr>
            <tr>{this.getField(this.state.cost)}</tr>
            <tr>{this.getField(this.state.dscript)}</tr>
          </tbody>
        </Table>
      </div>
    );
  }
```

```
 render() {
    return (
      <div>
        <Table className="table-hover">
          <tbody>
            <tr>{this.getField(this.state.loc)}</tr>
            <tr>{this.getField(this.state.begin_time)}</tr>
            <tr>{this.getField(this.state.end_time)}</tr>
            <tr>{this.getField(this.state.cost)}</tr>
            <tr>{this.getField(this.state.dscript)}</tr>
          </tbody>
        </Table>
      </div>
    );
  }
```

I wonder if a return object could be inject into a ```EventInfo``` component too.

Reply:
Never mind this - seeing your other comment now.